src/server/project.ts

```diff
@@ -1867,9 +1883,10 @@ namespace ts.server {
                 noLib: true,
                 diagnostics: false,
                 skipLibCheck: true,
+                preserveSymlinks: true,

```

Comment:
Without this, we don’t try to find the project reference source at the realpath at all because of https://github.com/microsoft/TypeScript/blob/34adaaf6aaebc430149fac78761590c3d02f8352/src/compiler/program.ts#L2345-L2351

It doesn’t seem like this should be dependent upon `preserveSymlinks` in this case.

Reply:
That is because only I’m preserveSymlinks we have to map d.ts that might’ve been symlink to output to ensure that we are mapping it to source ... if it’s not preserved the file shouldn’t be symlink but actual .d.ts so it avoids the extra work .. why should that impact auto import provider

Reply:
Consider [this test case](https://github.com/microsoft/TypeScript/pull/39679/files#diff-06221ae2c56604275205c5c7354cf6dc). `/a/package.json` has a dependency upon `b`, which is symlinked into `/a/node_modules/b`. The auto import provider sets its root files to the result of `resolveModuleName` for `"b"`, which is `/a/node_modules/b/out/index.d.ts`. Because `/b/tsconfig.json` is not built yet, that file does not exist. So when the auto import provider looks for the source of that .d.ts file to use instead, it first looks for `/a/node_modules/b/src/index.ts`. But because the project reference from `/a/tsconfig.json` to `/b/tsconfig.json` was resolved by its real path, the `mapFromToProjectReferenceRedirectSource` contains only real paths, so all lookups for `a/node_modules/b/*` will return undefined. Consequently, the auto-import provider is empty.

Reply:
Looking at how this works outside of AutoImportProviderProject, the resolution cache’s `resolveNamesWithLocalCache` does a `realpath` call on the resolved filename at the end that we don’t get from `ts.resolveTypeReferenceDirective`... I’ll try adding that to the auto import provider.

Reply:
Yep, that fixes it! Can you take another look?